rts/Sim/Units/UnitTypes/Factory.h

```diff
@@ -34,7 +34,7 @@ class CFactory : public CBuilding
 	/// supply the build piece to speed up
 	float3 CalcBuildPos(int buildPiece = -1);
 
-	void KillUnit(CUnit* attacker, bool selfDestruct, bool reclaimed);
+	void KillUnit(CUnit* attacker, bool selfDestruct, bool reclaimed, int weaponDefID = 0);
```

Comment:
Don't default to 0, this is because
1) that represents some actual weapon (see #509),
2) even if the above is fixed, now that every death type has its own constant there is always something appropriate to pass. There shouldn't be a case where a death was caused by some default/unknown weapon.

Reply:
Done.